packages/d3fc-webgl/README.md

```diff
@@ -1,16 +1,750 @@
+
+<a name="webglBaseAttribute_normalized" href="#webglBaseAttribute_normalized">#</a> *webglBaseAttribute*.**normalized**(*boolean*)
+
+If *boolean* is specified, sets the normalized property and returns this attribute builder. If *boolean* is not specified, returns the current value of normalized.
+
+The normalized property specifies whether integer data values should be normalized when being cast to a float, the default value is false.
+
+More information on how values are normalized can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/WebGLBuffer)
+
+<a name="webglBaseAttribute_stride" href="#webglBaseAttribute_stride">#</a> *webglBaseAttribute*.**stride**(*stride*)
+
+If *stride* is specified, sets the stride property and returns this attribute builder. If *stride* is not specified, returns the current stride.
+
+The stride property is used to specify the offset in bytes between the start of consecutive vertex attribute values, it must be a value from `0` (default) to `255`. By using the stride property it is possible to create a buffer of interleaved attribute values.
+
+<a name="webglBaseAttribute_offset" href="#webglBaseAttribute_offset">#</a> *webglBaseAttribute*.**offset**(*offset*)
+
+If *offset* is specified, sets the offset property and returns this attribute builder. If *offset* is not specified, returns the current offset.
+
+The offset property is used to specify the offset in bytes of the first value in the vertex attribute array. If set, the offset must be a multiple of the byte length of [type](#webglBaseAttribute_type).
+
+#### Uniform Builder
+
+<a name="webglUniform" href="#webglUniform">#</a> fc.**webglUniform**()
+
+Used to create a uniform value that is provided to every vertex.
+
+<a name="webglUniform_location" href="#webglUniform_location">#</a> *webglUniform*.**location**(*index*)
+
+If *index* is specified, sets the location property and returns this uniform builder. If *index* is not specified, returns the current location.
+
+The location property is used to specify the index location of the uniform being modified. The appropriate value can be found using [`WebGLRenderingContext.getUniformLocation()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/getUniformLocation).
+
+<a name="webglUniform_data" href="#webglUniform_data">#</a> *webglUniform*.**data**(*data*)
+
+If *data* is specified, sets the data property and returns this uniform builder. If *data* is not specified, returns the current data.
+
+The data property is used to set the value of the uniform, the value provided can either be a single value or an array with a maximum length of `4`.
+
+#### Buffer Builder
+
+<a name="webglBufferBuilder" href="#webglBufferBuilder">#</a> fc.**webglBufferBuilder**()
+
+Used to manage and execute multiple attribute and uniform builders.
+
+This component is useful for executing multiple builders that contain values to be passed to a single program.
```

Comment:
This component manages the mapping of attribute/uniform builders to their shader identifiers.